### PR TITLE
Validation

### PR DESCRIPTION
**Pull request by Swiddis**
### Description
Tracking PR for merging to 2.x

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/kotlin/org/opensearch/observability/validation/IntegrationValidator.kt [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    private val medeiaApi = MedeiaJacksonApi()
+
+    private fun loadSchema(): SchemaValidator {
+        val resource = javaClass.getResource("/schema/system/integration.schema")
+        resource ?: run {
+            log.error("failed to load integration schema resource")
+            throw IntegrationValidationException("failed to load integration schema resource")
+        }
```
Comment (derek-ho) [inline src/main/kotlin/org/opensearch/observability/validation/IntegrationValidator.kt]: Thinking about extensibility, do you want this to take in a file and be able to validate it? Then this class can be re-used to validate other things like index pattern, etc.

File: src/main/kotlin/org/opensearch/observability/validation/IntegrationValidator.kt [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    private fun invalid(message: String? = null, cause: Throwable? = null): IntegrationValidationException {
+        when {
+            message != null -> log.warn("validation failed: $message")
+            cause != null -> log.warn("validation failed due to exception: ${cause.javaClass}")
+            else -> log.error("validation failed for unspecified reason")
+        }
+        return IntegrationValidationException(message, cause)
+    }
+
+    fun validate(): Result<JsonNode> {
```
Comment (derek-ho) [inline src/main/kotlin/org/opensearch/observability/validation/IntegrationValidator.kt]: Where are these called?

File: src/main/resources/schema/system/integration.schema [hunk 3]
```diff
@@ -0,0 +1,139 @@
+{
```
Comment (derek-ho) [inline src/main/resources/schema/system/integration.schema]: this file already exists?

Comment (Swiddis) [reply src/main/resources/schema/system/integration.schema]: Yeah, had to copy it to the resources path to get it to work in the moment, will need to find an alternative way to get global paths later.

File: src/main/kotlin/org/opensearch/observability/validation/IntegrationValidator.kt (hunk 1 above)
Comment (Swiddis) [reply src/main/kotlin/org/opensearch/observability/validation/IntegrationValidator.kt]: Good idea! Will definitely have to incorporate it.

File: src/main/kotlin/org/opensearch/observability/validation/IntegrationValidator.kt (hunk 2 above)
Comment (Swiddis) [reply src/main/kotlin/org/opensearch/observability/validation/IntegrationValidator.kt]: Will be a part of the validation API when it's added

**Comment by Swiddis** [comment]
Close to move to upstream

**Pull request closed without merging**